operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -1,24 +1,46 @@
 package storage
 
 import (
+	"context"
 	"crypto/sha1"
 	"fmt"
 	"sort"
 
 	"github.com/ViaQ/logerr/v2/kverrors"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
 	"github.com/grafana/loki/operator/internal/manifests/storage"
+	"github.com/grafana/loki/operator/internal/status"
 )
 
 var hashSeparator = []byte(",")
 
-// ExtractSecret reads a k8s secret into a manifest object storage struct if valid.
-func ExtractSecret(s *corev1.Secret, secretType lokiv1.ObjectStorageSecretType) (*storage.Options, error) {
+func getSecret(ctx context.Context, k k8s.Client, stack *lokiv1.LokiStack) (*corev1.Secret, error) {
```

Comment:
Thinking about error handling, if we plan for this function to be used to fetch other secrets other than the object storage one then, we should also provide a way to configure the DegradedError struct. 

I am however more of the opinion of changing the name of this function to be more specific and not allude to the idea of reusing it when we need to get secrets from k8s. Instead, we should create a new function for getting new secrets out of k8s.